src/client/pmix_client.c

```diff
@@ -541,18 +548,104 @@ PMIX_EXPORT pmix_status_t PMIx_Abort(int flag, const char msg[],
      return PMIX_SUCCESS;
  }
 
+#if PMIX_HAVE_ZLIB
+static bool compress_string(char *instring,
+                            uint8_t **outbytes,
+                            size_t *nbytes)
+{
+    z_stream strm;
+    size_t len, outlen;
+    uint8_t *tmp, *ptr;
+    uint32_t inlen;
+
+    /* set default output */
+    *outbytes = NULL;
+
+    /* setup the stream */
+    inlen = strlen(instring);
+    memset (&strm, 0, sizeof (strm));
+    deflateInit (&strm, 9);
+
+    /* get an upper bound on the required output storage */
+    len = deflateBound(&strm, inlen);
+    if (NULL == (tmp = (uint8_t*)malloc(len))) {
+        *outbytes = NULL;
+        return false;
+    }
+    strm.next_in = (uint8_t*)instring;
+    strm.avail_in = strlen(instring);
+
+    /* allocating the upper bound guarantees zlib will
+     * always successfully compress into the available space */
+    strm.avail_out = len;
+    strm.next_out = tmp;
+
+    deflate (&strm, Z_FINISH);
+    deflateEnd (&strm);
+
+    /* allocate 4 bytes beyond the size reqd by zlib so we
+     * can pass the size of the uncompressed string to the
+     * decompress side */
+    outlen = len - strm.avail_out + 4;
```

Comment:
Instead of using `4` here can we use `sizeof(uint32_t)`? I know it will always be the same, but that would make the code a bit more clear for future selves. (same comment for the other `4` values below)